hazelcast/src/main/java/com/hazelcast/client/impl/connection/AddressProvider.java

```diff
@@ -29,9 +30,15 @@ public interface AddressProvider {
     /**
      * @return The possible member addresses to connect to.
      * @throws Exception when a remote service can not provide addressee
+     * @deprecated use {@link #loadAddresses(ClientConnectionProcessListener)} instead
      */
+    @Deprecated
     Addresses loadAddresses() throws Exception;
 
+    default Addresses loadAddresses(ClientConnectionProcessListener listener) throws Exception {
```

Comment:
Maybe instead of passing it as a parameter, we could a field for it in the address provider implementations and tweak the `loadAddress` as accordingly? I feel like it will be more clean this way

Reply:
Keeping as a parameter for now, as discussed in DM